azure-pipelines.yaml

```diff
@@ -83,37 +84,41 @@ stages:
           steps:
             - template: ado-pipeline/steps/wait.yaml
   
-    - stage: DeployAzureAgent
-      displayName: Deploy Azure DevOps Agent
-      dependsOn: Wait
-      jobs:
-        - job: Deploy_Terraform
-          displayName: Deploy DevOps Agent
-          pool:
-            vmImage: 'ubuntu-latest'
-          
-          steps:
-
-            - download: current
-              artifact: 'AzureDevOps-agent.tfplan'
-              displayName: 'Download AzureDevOps-agent.tfplan'
+  - stage: DeployAzureAgent
+    displayName: Deploy Azure DevOps Agent
+    dependsOn: Wait
+    jobs:
+      - job: Deploy_Terraform
+        displayName: Deploy DevOps Agent
+        pool:
+          vmImage: 'ubuntu-latest'
+        
+        steps:
 
-            - template: ado-pipeline/jobs/terraform-prep.yaml
-              parameters:
-                terraformVersion: $(tfVersion)
-                subscription: $(subscription)
-                resourceGroup: $(resourceGroup)
-                location: $(location)
-                storageAccount: $(storageAccount)
-                containerName: $(containerName)
-                stateKey: $(stateKey)-AzureDevOpsAgent.tfstate
-                workingDirectory: $(workDir)
+          - template: ado-pipeline/jobs/replace-token.yaml
```

Comment:
do we have this extension added in and would this be better?
https://marketplace.visualstudio.com/items?itemName=qetza.replacetokens